src/Tests/LinkEmbedFormatterTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\url_embed\Tests\LinkEmbedFormatterTest.
+ */
+
+namespace Drupal\url_embed\Tests; 
+
+use Drupal\link\Tests\LinkFieldTest;
+use Drupal\Component\Utility\SafeMarkup;
+use Drupal\Component\Utility\Unicode;
+use Drupal\Core\Url;
+use Drupal\link\LinkItemInterface;
+use Drupal\simpletest\WebTestBase;
+use Drupal\url_embed\Tests\UrlEmbedTestBase;
+/**
+ * Tests link field widgets and formatters.
+ *
+ * @group url_embed
+ */
+class LinkEmbedFormatterTest extends LinkFieldTest{
+
+
+  /**
+   * Tests the 'link_embed' formatter.
+   *
+   * This test is mostly the same as testLinkFormatter(), but they cannot be
+   * merged, since they involve different configuration and output.
+   */
+  function testLinkEmbedFormatter() {
+    $field_name = Unicode::strtolower($this->randomMachineName());
+    // Create a field with settings to validate.
+    $this->fieldStorage = entity_create('field_storage_config', array(
+      'field_name' => $field_name,
+      'entity_type' => 'entity_test',
+      'type' => 'link',
+      'cardinality' => 2,
+    ));
+    $this->fieldStorage->save();
+    entity_create('field_config', array(
+      'field_storage' => $this->fieldStorage,
+      'bundle' => 'entity_test',
+      'settings' => array(
+        'title' => DRUPAL_OPTIONAL,
+        'link_type' => LinkItemInterface::LINK_GENERIC,
+      ),
+    ))->save();
+    $display_options = array(
+      'type' => 'link_separate',
+      'label' => 'hidden',
+    );
+    entity_get_form_display('entity_test', 'entity_test', 'default')
+      ->setComponent($field_name, array(
+        'type' => 'link_default',
+      ))
+      ->save();
+    entity_get_display('entity_test', 'entity_test', 'full')
+      ->setComponent($field_name, $display_options)
+      ->save();
+
+    // Create an entity with one link field values:
+    // - The first field item uses a URL only.
+    // - The second field item uses a URL and link text.
+    // For consistency in assertion code below, the URL is assigned to the title
+    // variable for the first field.
+    $this->drupalGet('entity_test/add');
+    $url = UrlEmbedTestBase::getSampleUrl();
```

Comment:
Note that if you merge in the latest 8.x-1.x you can just call `UrlEmbedTestBase::FLICKR_URL` and `UrlEmbedTestBase::FLICKR_OUTPUT`.